de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/snippet/UiSnippetCustomRequirementsAttributeTable.java

```diff
@@ -0,0 +1,395 @@
+	
+	protected final String arrayInstanceID;
+
+	protected Set<RequirementType> requirementTypes = new HashSet<RequirementType>();
+	protected int maxNumberAttributes = 0;
+
+	protected EsfMarkerImageProvider emip = new EsfMarkerImageProvider();
+
+	protected TableViewerColumn colStatus = null;
+	protected List<TableViewerColumn> attColumns;
+	
+	protected boolean controlListnerActive = true;
+
+	/**
+	 * constructor of the abstract UI snippet array instance category table
+	 * instantiate the ID of the concept and the active category id
+	 * 
+	 * @param conceptId
+	 *            the id of the concept
+	 * @param categoryId
+	 *            the id of the category
+	 * @param arrayInstanceId
+	 *            the array instance id
+	 * @param activeForCategoryId
+	 *            the active for the category id
+	 * @param fullQualifiedCategoryId
+	 *            the full qualified category id
+	 * @param style
+	 *            The style as binary mask defining which buttons should be
+	 *            displayed with the given table (e.g. Add/Remove/Drill-Down)
+	 */
+	public UiSnippetCustomRequirementsAttributeTable(String conceptId, String categoryId, String arrayInstanceId,
+			String activeForCategoryId, String fullQualifiedCategoryId, int style) {
+		super(conceptId, categoryId, arrayInstanceId, activeForCategoryId, fullQualifiedCategoryId, style);
+		this.arrayInstanceID = arrayInstanceId;
+	}
+
+	/**
+	 * Implementation of the createTableColumns Method
+	 * 
+	 * @param editingDomain
+	 *            the editing domain which shall be used to provide the editing
+	 *            support in the columns
+	 */
+	protected void createTableColumns(EditingDomain editingDomain) {
+
+		// Status column
+		if (colStatus == null) {
+			colStatus = (TableViewerColumn) createDefaultColumn(COLUMN_TEXT_STATUS);
+
+			colStatus.setEditingSupport(createEditingSupport(editingDomain, categoryModel.getProperties()
+					.get(RequirementsAttributeLabelProvider.REQUIREMENT_STATUS_PROPERTY_NUMBER)));
+
+			colStatus.getColumn().setWidth(STATUS_COLUMN_WIDTH);
+
+			// initialize list for attribute column
+			attColumns = new ArrayList<>();
+		}
+
+		if (model instanceof CategoryAssignment) {
+
+			// Find all necessary requirement types
+			CategoryAssignmentHelper helper = new CategoryAssignmentHelper((CategoryAssignment) model);
+			ArrayInstance array = (ArrayInstance) helper.getPropertyInstance(arrayInstanceID);
+			List<APropertyInstance> arrayInstances = array.getArrayInstances();
+
+			// Some ugly casting is necessary here because beans cannot be used as the array
+			// property can be in different CAs
+			for (APropertyInstance arrayInstance : arrayInstances) {
+				CategoryAssignment reqObject = ((ComposedPropertyInstance) arrayInstance).getTypeInstance();
+				if (((Category) reqObject.getType()).getFullQualifiedName()
+						.equals(Requirement.FULL_QUALIFIED_CATEGORY_NAME)) {
+					RequirementType requirementType = (new Requirement(reqObject)).getReqType();

```

Comment:
Doesnt `new Requirement(reqObject).getReqType()` without the weird extra brackeing work?